keras2onnx/ke2onnx/upsample.py

```diff
@@ -47,10 +47,10 @@ def convert_keras_upsample(scope, operator, container, n_dims):
     # If channels_first is True, we don't need to permute the output of ONNX Upsample. Otherwise, similar to Crop's
     # conversion, a Transpose would be added.
     if channels_first:
-        apply_upsample(scope, input_tensor_name, operator.outputs[0].full_name, container, scales=scales)
+        apply_upsample(scope, [input_tensor_name], operator.outputs[0].full_name, container, scales=scales)
```

Comment:
Why there is a change here from input_tensor_name to [input_tensor_name]? do we change interface somewhere?

Reply:
The apply_upsample function in onnxconverter_common/onnx_ops.py requires a input_name list, as "scales" is now an additional input in opset 9.

Reply:
See https://github.com/onnx/onnxmltools/pull/281

Reply:
Reverted Upsample API changes in latest update.

Reply:
I am a little confused here. Previously this works without any changes on apply_upsample. Why we need make onnxmltools change to fix this? If we need this fix, why it works previously?

Reply:
There are two ways to do this: 
1. pass in input_name as a list containing a single string, where the upsample function uses that list as expected input.
2. pass in input_name as a string, where the upsample function accounts for this change in the function. The container.add_node function accounts for a string input and converts it to a list. 

Method 1 was implemented in the upsample function in apply_operation in ONNXMLTools (now onnx_ops in onnxconverter-common), and Method 2 was implemented in keras-onnx. To reconcile the difference, I originally aligned this version with method 1. After realizing most functions use method 2, I reverted the changes and aligned with the keras-onnx version. This required changing the original onnxmltools apply_upsample to require input_name as a string instead of a list. The code above is now outdated.

Reply:
Both methods work individually, but since we're moving all apply_ops to onnxconverter-common, using half of method 1 and half of method 2 will not work. The reason method 1 was even considered was due to the way the function was called in upsample.py from the coreml converter -- in opset 9, 'scale' became an additional input instead of an initializer, so in that case, a list of multiple string input names must be passed to container.add_node().

Reply:
Just checked, for keras opset 9, input_name is not a list. Did I miss something?